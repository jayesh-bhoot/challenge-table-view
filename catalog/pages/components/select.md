

``` react
state: {
  value: ''
}
---
<Select
  label='Select'
  value={state.value}
  items={['United States', 'Canada', 'Mexico']}
  onChange={e => setState({ value: e.target.value })}
/>
```

for more details check https://material-ui.com/components/selects/
