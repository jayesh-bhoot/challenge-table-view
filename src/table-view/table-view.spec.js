import React from 'react'
import { Button } from '@material-ui/core'
import { TableRow } from '@material-ui/core'

import TableView from './table-view'
import { default as Select } from '../select'

describe('Unfiltered TableView', () => {
  const props = {
    columns: [
      { label: 'Name', dataKey: 'name' },
      { label: 'Age', dataKey: 'age' },
      { label: 'Gender', dataKey: 'gender' }
    ],
    rows: [
      { name: 'Mike', age: 25, id: 1245, gender: 'Male' },
      { name: 'Jennifer', age: 30, id: 46874, gender: 'Female' },
      { name: 'Stella', age: 25, id: 46874, gender: 'Other' }
    ],
    filters: [
      { label: 'Age', dataKey: 'age', value: undefined },
      { label: 'Gender', dataKey: 'gender', value: [] }
    ]
  }

  test('renders a TableView', () => {
    const wrapper = shallow(<TableView {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should count 2 filter selects', () => {
    const wrapper = mount(<TableView {...props} />)
    expect(wrapper.find(Select)).toHaveLength(2)
  })

  test('should verify sane defaults for filter selects', () => {
    const wrapper = mount(<TableView {...props} />)
    wrapper.find(Select).forEach(s => expect(s.props().value).toHaveLength(0))
  })

  test('should count 0 clear filter button', () => {
    const wrapper = mount(<TableView {...props} />)
    expect(wrapper.find(Button)).toHaveLength(0)
  })

  test('should count 3 data rows + 1 header row', () => {
    const wrapper = mount(<TableView {...props} />)
    expect(wrapper.find(TableRow)).toHaveLength(4)
  })
})


describe('Filtered TableView', () => {
  const props = {
    columns: [
      { label: 'Name', dataKey: 'name' },
      { label: 'Age', dataKey: 'age' },
      { label: 'Gender', dataKey: 'gender' }
    ],
    rows: [
      { name: 'Mike', age: 25, id: 1245, gender: 'Male' },
      { name: 'Jennifer', age: 30, id: 46874, gender: 'Female' },
      { name: 'Stella', age: 25, id: 46874, gender: 'Other' }
    ],
    filters: [
      { label: 'Age', dataKey: 'age', value: [25] }
    ]
  }

  test('should count 1 clear filter button', () => {
    const wrapper = mount(<TableView {...props} />)
    expect(wrapper.find(Button)).toHaveLength(1)
  })

  test('should count 1 header plus 2 filtered data rows', () => {
    const wrapper = mount(<TableView {...props} />)
    expect(wrapper.find(TableRow)).toHaveLength(3)
  })

  test('should verify 1 value in filter select', () => {
    const wrapper = mount(<TableView {...props} />)
    expect(wrapper.find(Select).props().value).toEqual([25])
  })
})
