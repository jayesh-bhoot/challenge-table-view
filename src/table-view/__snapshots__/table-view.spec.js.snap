// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unfiltered TableView renders a TableView 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Select
      items={
        Array [
          25,
          30,
        ]
      }
      key="age"
      label="Age"
      multiple={true}
      onChange={[Function]}
      value={Array []}
    />
    <Select
      items={
        Array [
          "Female",
          "Male",
          "Other",
        ]
      }
      key="gender"
      label="Gender"
      multiple={true}
      onChange={[Function]}
      value={Array []}
    />
  </div>
  <div>
    <WithStyles(Table)
      columns={
        Array [
          Object {
            "dataKey": "name",
            "label": "Name",
          },
          Object {
            "dataKey": "age",
            "label": "Age",
          },
          Object {
            "dataKey": "gender",
            "label": "Gender",
          },
        ]
      }
      rows={
        Array [
          Object {
            "age": 25,
            "gender": "Male",
            "id": 1245,
            "name": "Mike",
          },
          Object {
            "age": 30,
            "gender": "Female",
            "id": 46874,
            "name": "Jennifer",
          },
          Object {
            "age": 25,
            "gender": "Other",
            "id": 46874,
            "name": "Stella",
          },
        ]
      }
    />
  </div>
</div>
`;
